File: stock-market-service/src/main/java/rs/edu/raf/dto/StockWeeklyDTO.java
package rs.edu.raf.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

public record StockWeeklyDTO(
    @JsonProperty("Meta Data")
    Map<String, String> metaData,
    @JsonProperty("Weekly Time Series")
    Map<String, StockHistoryInfoDTO> timeSeriesWeekly
) { }

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/dto/StockWeeklyDTO.kt
package org.example.stockmarketservicekotlin.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class StockWeeklyDTO(
    @JsonProperty("Meta Data")
    val metaData: Map<String, String>,
    @JsonProperty("Weekly Time Series")
    val timeSeriesWeekly: Map<String, StockHistoryInfoDTO>
)
